gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingSource.java

```diff
@@ -114,28 +121,30 @@ static public Consumer getKafkaConsumer(State state) {
 
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
-      Consumer<String, byte[]> consumer = getKafkaConsumer(state);
-      LOG.debug("Consumer is {}", consumer);
-      String topic = state.getProp(TOPIC_WHITELIST); // TODO: fix this to use the new API when KafkaWrapper is fixed
-      List<WorkUnit> workUnits = new ArrayList<WorkUnit>();
-      List<PartitionInfo> topicPartitions;
-      topicPartitions = consumer.partitionsFor(topic);
-      LOG.debug("Partition count is {}", topicPartitions.size());
-      for (PartitionInfo topicPartition : topicPartitions){
-        Extract extract = this.createExtract(DEFAULT_TABLE_TYPE, DEFAULT_NAMESPACE_NAME, topicPartition.topic());
-        LOG.debug("Partition info is {}", topicPartition);
-        WorkUnit workUnit = WorkUnit.create(extract);
-        setTopicNameInState(workUnit, topicPartition.topic());
-        workUnit.setProp(ConfigurationKeys.EXTRACT_TABLE_NAME_KEY, topicPartition.topic());
-        setPartitionId(workUnit, topicPartition.partition());
-        workUnits.add(workUnit);
-      }
-      return workUnits;
+    Config config = ConfigUtils.propertiesToConfig(state.getProperties());
+    Consumer<String, byte[]> consumer = getKafkaConsumer(state);
```

Comment:
You can replace this with getKafkaConsumer(config)  and change getKafkaConsumer to take in Config as parameter